src/screens/authentication/SocialNetworks/GoogleButton/index.js

```diff
@@ -1,14 +1,41 @@
-import React from 'react'
+import React, { useState } from 'react'
+import { ActivityIndicator } from 'react-native'
 import Icon from 'react-native-vector-icons/FontAwesome5'
-import { Text } from '~/components'
+import NavigationService from '~/navigationService'
+import * as DatabaseService from '~/databaseService'
 import SocialNetworkButton from '../SocialNetworkButton'
-import styles from './styles'
-
-const GoogleButton = ({ style, onPress }) => (
-  <SocialNetworkButton style={[styles.button, style]} onPress={onPress}>
-    <Icon style={styles.icon} name='google' size={15} color='white' />
-    <Text style={styles.text}>Ingresá con tu cuenta de Google</Text>
-  </SocialNetworkButton>
-)
+import * as AuthenticationService from '~/authenticationService'
+import styles, { ICON_SIZE } from './styles'
+
+const GoogleButton = ({ style }) => {
+  const [signing, setSigning] = useState(false)
+
+  const onGoogleButtonPress = async () => {
+    setSigning(true)
+    try {
+      const firebaseCredential = await AuthenticationService.googleLogin()
+      const username = await DatabaseService.queryUsernameFromEmail(firebaseCredential.user.email)
+
+      setSigning(false)
+      if (username) {
+        NavigationService.navigate('MainApp')
+      } else {
+        NavigationService.navigate('UsernameRequest')
+      }
+    } catch (error) {
+      setSigning(false)
+    }
```

Comment:
Avoid repeating `setSigning(false)` by calling it in the `finally` block:
```javascript
try {
  // something that may fail
} catch (err) {
  // do something with the error
} finally {
  // do something whether an error was thrown or not
}
```
Just as you can omit the `finally` block, you can also omit the `catch` block. The following is valid:
```javascript
try {
  // something that may fail
} finally {
  // do something whether an error was thrown or not
}
```

Reply:
Repetition was mainly because NavigationService prompts the component to be unmounted and a warning appears showing that "setting state in an unmounted component is a no-op". So it was necessary to set the state before navigating. I can diminish repetition with a boolean workaround though.

Reply:
The solution here, in my opinion, would be extracting the variable and the navigation outside of the `try/catch`:
```javascript
setSigning(true)

let username
try {
  const firebaseCredential = await AuthenticationService.googleLogin()
  username = await DatabaseService.queryUsernameFromEmail(firebaseCredential.user.email)
} finally {
  setSigning(false)
}

if (username) {
  NavigationService.navigate('MainApp')
} else {
  NavigationService.navigate('UsernameRequest')
}
```